patchwork-capabilities/src/main/java/com/patchworkmc/impl/capability/BaseCapabilityProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.impl.capability;
+
+import javax.annotation.Nullable;
+
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.capabilities.CapabilityDispatcher;
+import net.minecraftforge.common.capabilities.CapabilityProvider;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+
+public class BaseCapabilityProvider<T> extends CapabilityProvider<T> {
+	private final T provider;
+
+	public BaseCapabilityProvider(Class<T> baseClass, T provider) {
+		super(baseClass);
+		this.provider = provider;
+	}
+
+	@Override
+	public void gatherCapabilities(@Nullable ICapabilityProvider parent) {
+		AttachCapabilitiesEvent<T> event = new AttachCapabilitiesEvent<>(baseClass, provider);
+		MinecraftForge.EVENT_BUS.post(event);
+
+		capabilities = !event.getCapabilities().isEmpty() || parent != null ? new CapabilityDispatcher(event.getCapabilities(), event.getListeners(), parent) : null;
```

Comment:
Minor nit that you're free to ignore, I'd personally prefer an if statement here over a ternary. (are there any case where this will be called when capabilities != null?)

Reply:
I agree, this line is a bit l o n g